### PR TITLE
Update to stable 1.12 mappings

### PR DESCRIPTION
**Pull request by Diamondman121314**
Updated everything to more recent versions.

Also, I'm not sure if it's possible for me to create a new branch as this is the first time I'm doing something like this.

### CODE REVIEW

**Review by exaskye — APPROVED** [review]
This is really just a mappings update, not really a 1.12.2 update. Concrete works as-is on 1.12.2, which is why it hasn't been updated.

But it's fine, I guess? 1.12.2 is being sunset so I guess there's no harm in updating to the final release and mappings (other than the fact all our build infra is gone now and if this is merged nothing will *really* happen anyway...)

**Comment by Diamondman121314** [comment]
Ah, I didn't realize this. My main reason for updating was that, I was updating another mod that used this and when in-game, it caused a crash because of an old mapping. If it doesn't get merged, that's fine.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.5-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.3-all.zip
```
Comment (falkreon) [inline gradle/wrapper/gradle-wrapper.properties]: Changing the gradle wrapper is outside the scope of this PR

**Comment by exaskye** [comment]
Yeah, it will bite you if using Concrete as source. I tended to fix that by using the older mappings since a lot of the changed names were kind of nitpicky and I had gotten used to the weird old names from being a modder for so long.

**Review by falkreon — CHANGES_REQUESTED** [review]
I only see a wrapper update in gradle.properties, but the wrapper itself is missing from the PR. That makes me worry that the wrapper isn't actually *in* the repo because of gitignore.

Either way, we definitely don't want to include it in the PR, it's a separate problem.

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (Diamondman121314) [reply gradle/wrapper/gradle-wrapper.properties]: I didn't realize that it got updated, fixed.

**Comment by exaskye** [comment]
gradle-wrapper.jar and the gradlew/gradlew.bat scripts have not changed in a very, very long time. Only the URL needs to be updated.

But yeah, I suppose it is out-of-scope.

**Review by falkreon — APPROVED** [review]

**Comment by exaskye** [comment]
(We don't need to bother more people with a PR on a 85% dead project. 2 approves is good enough for me; let me sort out the branch nonsense.)

**Comment by exaskye** [comment]
...Why did that merge into 1.12.1.

GitHub pls

**Comment by Diamondman121314** [comment]
I'm not sure if this is affected by what you said earlier, but could this go on maven? If not, all good then.

**Comment by exaskye** [comment]
Our CI has been shuttered and that managed our Maven; I can do a manual push.

**Comment by exaskye** [comment]
The Kotlin update means the Gradle wrapper update is mandatory. I don't think the Kotlin support ever worked, though, honestly.

**Comment by exaskye** [comment]
Er, wait, this is Concrete. I mean *Skeleton's* Kotlin support. Concrete has a Kotlin module. It's been so long since I worked on this stuff <.<

**Comment by falkreon** [comment]
That's what I get for approving without testing X) Ah well. We'll get it working. Thanks for the PR Diamondman!

**Comment by Diamondman121314** [comment]
I think I only updated that because IntelliJ was complaining that it wasn't the same version that the plugin was. I did a bit of this yesterday and forgot until now to create this PR.

**Comment by exaskye** [comment]
0.4.46 has been manually published.

**Comment by exaskye** [comment]
(It probably works!)

**Comment by jamierocks** [comment]
Wow, didn't expect notifications from here 🎉

**Pull request merged**